### DISCUSSION TITLE
Upgrade path from wifi-desk-controller to upsy-desky

### DISCUSSION THREAD

**Discussion by rohankapoorcom** [tjhorner/wifi-desk-controller#22] [Support] [answered]
Hi TJ - is there a path to upgrade the software on this wifi-desk-controller to the upsy-desky version or would you recommend upgrading the hardware instead?

Is there any major reason to upgrade in your opinion?

Thanks!

**Comment by tjhorner** [comment, marked as answer]
Hi Rohan! I can look into porting some of the Upsy Desky features to the firmware here — what would you like to see specifically?

The biggest reason to upgrade to the Upsy Desky would have to be the fact that it uses an ESP32 module instead of an ESP8266 board. The most important reason for this change is that the ESP32 has [more hardware UART peripherals](https://esphome.io/components/uart.html#hardware-uarts), so the MCU can spend less time decoding UART signals, and the decoded signals will be more accurate.

Since the ESP32 also has Bluetooth functionality, it can act as a [Bluetooth proxy](https://esphome.io/components/bluetooth_proxy.html) to extend the Bluetooth range of your Home Assistant instance. You can also use any of the various other [ESPHome Bluetooth components](https://esphome.io/#bluetooth-low-energy-ble) if they are useful to you. It also generally has better specs — [here's a nice comparison table I found](https://makeradvisor.com/esp32-vs-esp8266/).

The board was entirely redesigned to be fully integrated (rather than relying on a dev board) and the enclosure was redesigned as well with a better way to mount the board and improved fit-and-finish. There are other differences as well (USB-C, extra power protection, logic level shifter at the RJ45 ports) but they are minor.

If you do decide to upgrade, [let me know](mailto:me@tjhorner.dev) and I can give you a hefty discount since you already have a Wi-Fi Desk Controller.

Hope this helps!

**Reply by rohankapoorcom** [reply]
Hey TJ - I was mainly interested in the stability improvements. I have an Uplift V2 and have this irritating problem that every time the desk power is cycled for any reason, I have to get under the desk and disconnect the wifi desk controller and reconnect it to both the desk and the buttons on the desk otherwise nothing moves and it's locked into the lowest position. It sounds like that is a hardware fix and not a software fix of the new firmware?

Is there any firmware specific benefits to the new version? Otherwise, it sounds like upgrading the hardware may be what I want to do.

Thanks for the response!

**Reply by tjhorner** [reply]
Oh, yes — it's been such a long time since I've thought of those issues I forgot to mention that they are fixed 😅. That's indeed a hardware fix, yes. The issue you are describing as well as #6 are both fixed as they had the same root cause. If you are interested in the technicals: one of the GPIOs (D8/GPIO15) used for communication with the desk was also used for CS, so would fail to boot if pulled high during power-on.

**Reply by rohankapoorcom** [reply]
Ah perfect, sounds like it's time to upgrade then. Let me send you an email!

You don't happen to have a variant that uses ethernet for communication do you? I'd be interested in the bluetooth proxy functionality, but I don't like my bluetooth proxies to be on wifi.
